adapters/grid/grid.go

```diff
@@ -69,16 +97,222 @@ func setImpExtData(imp openrtb2.Imp) openrtb2.Imp {
 	return imp
 }
 
+func mixKeywords(keywordsString string, keywords map[string]interface{}) {
+	var ortb2Array []interface{}
+
+	keywordsArr := strings.Split(keywordsString, ",")
+
+	if len(keywordsArr) > 0 {
+		keywordsInt := make([]interface{}, len(keywordsArr))
+		for i, v := range keywordsArr {
+			keywordsInt[i] = v
+		}
+		ortb2Keywords := map[string]interface{}{
+			"name":     "keywords",
+			"keywords": keywordsInt,
+		}
+		if keywords["ortb2"] == nil {
+			ortb2Array = make([]interface{}, 0)
+		} else {
+			ortb2Array = keywords["ortb2"].([]interface{})
+		}
+		ortb2Array = append(ortb2Array, ortb2Keywords)
+		keywords["ortb2"] = ortb2Array
+	}
+}
+
+func mergeWithReqExtKeywords(extKeywords map[string]interface{}, request *openrtb2.BidRequest) error {
+	var reqExt ReqExt
+	var reqExtKeywords map[string]interface{}
+
+	if err := json.Unmarshal(request.Ext, &reqExt); err != nil {
+		return err
+	}
+	if reqExt.Keywords != nil {
+		json.Unmarshal(reqExt.Keywords, &reqExtKeywords)
```

Comment:
We realize an error scenario here is unlikely but we believe it's better to try to check for it and handle it